lib/c/c_ConsumerConfiguration.cc

```diff
@@ -250,3 +251,36 @@ pulsar_consumer_regex_subscription_mode pulsar_consumer_configuration_get_regex_
     return (pulsar_consumer_regex_subscription_mode)
         consumer_configuration->consumerConfiguration.getRegexSubscriptionMode();
 }
+
+void pulsar_consumer_configuration_set_dlq_policy(pulsar_consumer_configuration_t *consumer_configuration,
+                                                  const char *dead_letter_topic, int max_redeliver_count,
+                                                  const char *initial_subscription_name) {
+    auto policyBuilder = pulsar::DeadLetterPolicyBuilder().maxRedeliverCount(max_redeliver_count);
+    if (dead_letter_topic != nullptr) {
+        policyBuilder.deadLetterTopic(dead_letter_topic);
+    }
+    if (initial_subscription_name != nullptr) {
+        policyBuilder.initialSubscriptionName(initial_subscription_name);
+    }
+    consumer_configuration->consumerConfiguration.setDeadLetterPolicy(policyBuilder.build());
+}
```

Comment:
It seems that this API isn't exposed at all. I don't think we need this API. The following API is enough.